core/src/test/java/org/testcontainers/junit/DependenciesTest.java

```diff
@@ -0,0 +1,116 @@
+package org.testcontainers.junit;
+
+import lombok.Getter;
+import org.junit.Test;
+import org.rnorth.visibleassertions.VisibleAssertions;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.startupcheck.OneShotStartupCheckStrategy;
+import org.testcontainers.lifecycle.Startable;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class DependenciesTest {
+
+    @Test
+    public void shouldWorkWithSimpleDependency() {
+        InvocationCountingStartable startable = new InvocationCountingStartable();
+
+        try (
+            GenericContainer container = new GenericContainer()
+                .withStartupCheckStrategy(new OneShotStartupCheckStrategy())
+                .dependsOn(startable)
+        ) {
+            container.start();
+        }
+
+        VisibleAssertions.assertEquals("Started once", 1, startable.getStartInvocationCount().intValue());
+        VisibleAssertions.assertEquals("Does not trigger .stop()", 0, startable.getStopInvocationCount().intValue());
```

Comment:
I'm sure there's a good reason, but why do we not propagate a `stop()` call through the dependencies?

Reply:
because there is no guarantee that the dependencies will be a part of the same "group".

We may introduce `deepStop` in future

Reply:
I think a `deepStop` would be useful for when you know these are the semantics you want. I end up having to manually create one in several projects. Thoughts on adding an API?

Btw, are both `stop` and `start` required to be idempotent? I think the current implementation of `deepStart` depends on this.